internal/local/structure_v2.go

```diff
@@ -331,3 +328,228 @@ func parseSync(rootDir string) (*SyncStructure, error) {
+		Location:              a.Location(),
+		DeploymentModel:       a.DeploymentModel(),
+		Environment:           a.Environment,
+		AllowedRequestOrigins: a.AllowedRequestOrigins,
+	}
+	return json.Marshal(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a AppDataV2) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	if err := writeGraphQL(rootDir, a.GraphQL); err != nil {
+		return err
+	}
+	if err := writeServices(rootDir, a.Services); err != nil {
+		return err
+	}
+	if err := writeFunctionsV2(rootDir, a.Functions); err != nil {
+		return err
+	}
+	if err := writeAuth(rootDir, a.Auth); err != nil {
+		return err
+	}
+	if err := writeSync(rootDir, a.Sync); err != nil {
+		return err
+	}
+	if err := writeDataSources(rootDir, a.DataSources); err != nil {
+		return err
+	}
+	if err := writeHTTPEndpoints(rootDir, a.HTTPEndpoints); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV2(rootDir string, functions *FunctionsStructure) error {
+	if functions != nil {
+		for _, function := range functions.Configs {
+			nameFunc := function["name"].(string)
+			dataFunc, err := MarshalJSON(function)
+			if err != nil {
+				return err
+			}
+			dataFunc = append(dataFunc, "\n"...)
+			err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunc, FileConfig.String()), 0666, bytes.NewReader(dataFunc))
+			if err != nil {
+				return err
+			}
+			dataSrc := functions.Sources[nameFunc]
+			err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunc, FileSource.String()), 0666, bytes.NewReader([]byte(dataSrc)))
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		err := WriteFile(filepath.Join(rootDir, NameFunctions, FileConfig.String()), 0666, bytes.NewReader([]byte("[]\n")))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeAuth(rootDir string, auth *AuthStructure) error {
+	if auth != nil && auth.Providers != nil {
+		dataProvider, err := MarshalJSON(auth.Providers)
+		if err != nil {
+			return err
+		}
+		dataProvider = append(dataProvider, "\n"...)
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileProviders.String()), 0666, bytes.NewReader(dataProvider))
+		if err != nil {
+			return err
+		}
+	} else {
+		err := WriteFile(filepath.Join(rootDir, NameAuth, FileProviders.String()), 0666, bytes.NewReader([]byte("{}\n")))
+		if err != nil {
+			return err
+		}
+	}
+	if auth != nil && auth.CustomUserData != nil {
+		dataCustomUser, err := MarshalJSON(auth.CustomUserData)
+		if err != nil {
+			return err
+		}
+		dataCustomUser = append(dataCustomUser, "\n"...)
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileCustomUserData.String()), 0666, bytes.NewReader(dataCustomUser))
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
just to retiterate, ideally these `else` blocks go away, and we simply just write what the app has

and now that i see this more and more, maybe it makes sense to instead of pushing that logic to `app init`/`app create`, but rather to the `NewApp`/`AsApp` constructors?

Reply:
Yeah, I'm gonna go in this direction. I didn't like putting defaults into the cmd (since they weren't cmd specific defaults)